### PR TITLE
Add utf8 from latin1 length function

### PR DESCRIPTION
**Pull request by Nick-Nuon**
Still a draft. did little more than copy/paste what's on the AVX blog post to the right spot, but working on it

### CODE REVIEW

**Comment by lemire** [comment]
@Nick-Nuon I patched a conflict on your PR.

**Comment by lemire** [comment]
@Nick-Nuon What is the state of this PR, do you recommend merging?

File: src/icelake/implementation.cpp [hunk 1]
```diff
@@ -1156,7 +1156,64 @@ simdutf_warn_unused size_t implementation::utf32_length_from_latin1(size_t lengt
 }
 
 simdutf_warn_unused size_t implementation::utf8_length_from_latin1(const char * input, size_t length) const noexcept {
-  return scalar::latin1::utf8_length_from_latin1(input,length);
+  const uint8_t *str = reinterpret_cast<const uint8_t *>(input);
+  size_t answer = length / sizeof(__m512i) * sizeof(__m512i);
+  size_t i = 0;
+  char v_0xFF = static_cast<unsigned char>(0xFF);
```
Comment (lemire) [inline src/icelake/implementation.cpp:1162]: This is weird code btw. It is correct but a bit strange.

**Comment by Nick-Nuon** [comment]
> @Nick-Nuon What is the state of this PR, do you recommend merging?

Five minutes ago, I would have said yes, but now with the strange code, I am not sure. 
 
 IIRC, one of the tests was complaining and that was something of a stopgap measure. 
 I wasn't sure how important elegance was so I moved on.

**Comment by lemire** [comment]
>  now with the strange code, I am not sure.

Well, it is a small thing, just do

`unsigned char v_ff = 0xff;`

or something of the sort. There might be an even cleaner solution.

**Comment by lemire** [comment]
Ok. So here, just so we are clear, you are unrolling some loops by a factor of four, which means that we expect to get inputs spanning more than 4*64 bytes (256 bytes). That is fine. We just must be aware of that.

**Comment by lemire** [comment]
@Nick-Nuon Merging.

**Pull request merged**